### PR TITLE
Add HLSL file support to amdllpc

### PR DESCRIPTION
**Pull request by LLJJDD**
Allow amdllpc to compile hlsl source file with setting entry point correctly.

### CODE REVIEW

File: tool/amdllpc.cpp [hunk 1]
```diff
@@ -624,11 +624,13 @@ static Result CompileGlsl(
         void* pProgram = nullptr;
         const char* pLog = nullptr;
         int compileOption = SpvGenOptionDefaultDesktop | SpvGenOptionVulkanRules | SpvGenOptionDebug;
+        compileOption |= (isHlsl) ? SpvGenOptionReadHlsl : 0;
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: (isHlsl) -> isHlsl

File: tool/amdllpc.cpp [hunk 2]
```diff
@@ -624,11 +624,13 @@ static Result CompileGlsl(
...
+        const char* entryPoints[] = { EntryTarget.c_str() };
         bool compileResult = spvCompileAndLinkProgramEx(1,
                                                         &lang,
                                                         &sourceStringCount,
                                                         sourceList,
-                                                        nullptr,
+                                                        (isHlsl) ? entryPoints : nullptr,
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: (isHlsl) -> isHlsl

**Comment by amdvlk-admin** [comment]
### Test summary for commit 0478f0768b0a33d232db04de1530a5c1e6b42fb1
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 9a1f99f049f3b417242dc7c09b86c52fd7d9ceaf</li>
<li>SPVGEN: dev 64013f150c2d41965a3d36fa159b55fba52b727a</li>
<li>XGL: dev 4d91a4045078cf6884ee05007eac9eec2b881dbb</li>
<li>LLVM: amd-vulkan-dev 4305f776f1207189667a383e22661ccd1e99cd25</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 900/900 (100.0%)</li>
<li>Failed: 0/900 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 294162/294162 (100.0%)</li>
<li>Failed: 0/294162 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366303/366303 (100.0%)</li>
<li>Failed: 0/366303 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit b09f5aab7d87aee9b6362cc3773f6e313dee8071
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 9a1f99f049f3b417242dc7c09b86c52fd7d9ceaf</li>
<li>SPVGEN: dev 64013f150c2d41965a3d36fa159b55fba52b727a</li>
<li>XGL: dev 4d91a4045078cf6884ee05007eac9eec2b881dbb</li>
<li>LLVM: amd-vulkan-dev 4305f776f1207189667a383e22661ccd1e99cd25</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 900/900 (100.0%)</li>
<li>Failed: 0/900 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 294161/294162 (100.0%)</li>
<li><details><summary>Failed: 1/294162 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.image.swapchain_mutable.xlib.2d.r8g8b8a8_unorm_r8g8b8a8_snorm_clear_copy_format_list
Stack trace: Failed to open display: '' at tcuLnxX11.cpp:67
```
</p></details></li>


</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366303/366303 (100.0%)</li>
<li>Failed: 0/366303 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 7013f0de494a9c89c6ef6f50047f8ffa27e67c10
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 9a1f99f049f3b417242dc7c09b86c52fd7d9ceaf</li>
<li>SPVGEN: dev 64013f150c2d41965a3d36fa159b55fba52b727a</li>
<li>XGL: dev 4d91a4045078cf6884ee05007eac9eec2b881dbb</li>
<li>LLVM: amd-vulkan-dev b9910c8bfcccc63c07c959963fa567120d11f024</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 900/900 (100.0%)</li>
<li>Failed: 0/900 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 294161/294162 (100.0%)</li>
<li><details><summary>Failed: 1/294162 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.image.swapchain_mutable.xlib.2d.r8g8b8a8_unorm_r8g8b8a8_snorm_clear_copy_format_list
Stack trace: Failed to open display: '' at tcuLnxX11.cpp:67
```
</p></details></li>


</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366303/366303 (100.0%)</li>
<li>Failed: 0/366303 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 7013f0de494a9c89c6ef6f50047f8ffa27e67c10
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 9a1f99f049f3b417242dc7c09b86c52fd7d9ceaf</li>
<li>SPVGEN: dev 64013f150c2d41965a3d36fa159b55fba52b727a</li>
<li>XGL: dev 4d91a4045078cf6884ee05007eac9eec2b881dbb</li>
<li>LLVM: amd-vulkan-dev b9910c8bfcccc63c07c959963fa567120d11f024</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 900/900 (100.0%)</li>
<li>Failed: 0/900 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 366303/366303 (100.0%)</li>
<li>Failed: 0/366303 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366303/366303 (100.0%)</li>
<li>Failed: 0/366303 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**